### DISCUSSION TITLE
AGPL许可证问题

### DISCUSSION THREAD

**Discussion by ijry** [pipipi-pikachu/PPTist#246] [Q&A] [answered]
引用该软件(PPTist)具有传染性，需要开源所有项目代码吗？或者如果只是以iframe交互的方式引用呢？

**Comment by pipipi-pikachu** [comment]
理论上应该是的，可以看看详细的协议：https://www.gnu.org/licenses/agpl-3.0.en.html

**Comment by pipipi-pikachu** [comment, marked as answer]
但是事实上不会要求那么严格（就我个人而言），我更加关注的是：
1. 保留原始作者/版权信息；
2. 将相关部分二开后的代码继续以AGPL协议开源，但不必原封不动的开源，允许抹去一些较私密的数据信息后单独发布开源版本；

基于上面的要求，是否iframe其实也无所谓了，因为你只需要将`相关部分`的代码进行分发/开源，具体来说，假设你基于PPTist做了一个AI生成幻灯片的项目，这个项目有‘资源管理’、‘幻灯片编辑’、‘服务端’这三套代码组成，那么只需要将用到了PPTist的那套代码分发/开源即可。

> 注：以上只是我个人对此项目的要求，不代表AGPL-V3协议本身（协议本身要严格得多）。

**Reply by ijry** [reply]
好的谢谢！

**Comment by pipipi-pikachu** [comment]
最近发现了某些人（公司）的恶劣行径，打着遵循AGPL-3的旗子，实际上：
1. 假开源：仅开源产品官网代码、仅开源压缩打包后的代码。
2. 产品页面不主动提供开源/获取源代码链接。
3. 无视PPTist版权声明，声称版权归其自主所有。
4. 将PPTist的Vue代码修改为其他框架代码后取消开源，自称自主研发。
5. 以AGPL-3开源后向其他人提供代码的商用授权服务（最骚的操作）。

鉴于以上恶意行为，考虑这类人和公司可能会拿我上面的帖子说事，避免纠纷，故在此声明，上面原本我对此协议的宽松要求说法全部作废。要求**必须、完全、严格、丝毫不差地**遵循AGPL-3.0协议（注：本帖题主和我亲自在邮件中回复过的除外，但前提是不存在上述恶劣行为）。

多说几句，开源这件事本身至少应该是一个相互信任的过程，拿着别人的信任砖空子（严格来说不叫砖空子，应该叫恶意欺骗）这种行为令我作呕。
